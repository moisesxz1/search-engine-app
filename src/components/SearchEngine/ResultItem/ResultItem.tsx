import React from 'react';
import PropTypes from 'prop-types';

import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';

interface ResultItemProps {
  name: string;
}
const ResultItem: React.FC<ResultItemProps> = (props) => {
  const { name } = props;

  return (
    <div>
      <ListItem disablePadding>
        <ListItemText primary={name} />
      </ListItem>
    </div>
  );
};

ResultItem.propTypes = {
  name: PropTypes.string.isRequired,
};

export default ResultItem;
