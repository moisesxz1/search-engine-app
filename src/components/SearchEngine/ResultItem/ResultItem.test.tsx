import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import ResultItem from './ResultItem';

describe('Tests on the <ResultItem /> Component', () => {
  test('It should render <ResultItem /> correctly', () => {
    const wrapper = shallow(<ResultItem name='Coffee' />);

    expect(wrapper).toMatchSnapshot();
  });

  // test('It should render the name prop correctly', () => {
  //   const name = 'Coffee';

  //   const wrapper = shallow(<ResultItem name={name} />);

  //   expect(wrapper.find('ForwardRef').length).toEqual(1);
  // });
});
