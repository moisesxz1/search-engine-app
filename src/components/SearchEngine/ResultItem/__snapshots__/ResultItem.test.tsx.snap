// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on the <ResultItem /> Component It should render <ResultItem /> correctly 1`] = `
<div>
  <ForwardRef(ListItem)
    disablePadding={true}
  >
    <ForwardRef(ListItemText)
      primary="Coffee"
    />
  </ForwardRef(ListItem)>
</div>
`;
